packages/nimble-components/src/text-area/styles.ts

```diff
@@ -107,6 +127,45 @@ export const styles = css`
     :host([resize='vertical']) .control {
         resize: vertical;
     }
+
+    .container::after {
+        content: ' ';
+        position: absolute;
+        bottom: calc(-1 * ${borderWidth});
+        width: 0px;
+        height: 0px;
+        border-bottom: ${borderHoverColor}
+            var(--ni-private-hover-indicator-width) solid;
+        transition: width ${smallDelay} ease-in;
+    }
+
+    @media (prefers-reduced-motion) {
+        .container::after {
+            transition-duration: 0s;
+        }
+    }
+
+    :host([error-visible]) .container::after {
+        border-bottom-color: ${failColor};
+    }
+
+    :host(:hover) .container::after {
+        width: 100%;
+    }
+
+    :host([disabled]:hover) .container::after,
+    :host([readonly]:hover) .container::after {
+        width: 0px;
+    }
+
+    :host([error-visible]) .error-icon {
+        position: absolute;
+        top: calc(${standardPadding} / 2);
+        right: calc(${standardPadding} / 2);
+    }
+    :host([error-visible].vert-scrollbar) .error-icon {
+        right: calc(${standardPadding} / 2 + var(--ni-private-scrollbar-width));
```

Comment:
On macOS browsers I'm seeing that the text can overlap the error icon if the text takes the full width of the line. I think I'm only seeing this when the vertical scrollbar is visible.
<img width="253" alt="image" src="https://user-images.githubusercontent.com/10500124/226383542-a3013ac0-10b8-456f-af1f-bd5a4d955d9b.png">

Is it possible this could be related to the scrollbar only being visible while it's in use? Or the scrollbar having different widths on different platforms?

Reply:
On Android
![Screenshot_20230328-080128](https://user-images.githubusercontent.com/1588923/228244820-3c9c560d-479a-4a55-a325-c2d232b9b8c8.png)

Reply:
I wasn't able to reproduce the issue Jesse saw on Mac -- my scrollbars looked different and were always present if the text overflowed. However, I've taken a different approach that calculates the scrollbar width based on `clientWidth` and `offsetWidth`. I update `--ni-private-scrollbar-width` on the element instead of toggling the `vert-scrollbar` class. Hopefully this will also address the issues on Mac and Android.

As for the other overlap cases Milan pointed out, I've set a minimum height and width as was suggested.

Reply:
Just to chime in, I also can't reproduce what Jesse on Mac. I'm on OS 13.2.1 using Safari 16.3.

Reply:
It turns out that what Jesse was seeing was due to a system setting that handles trackpad use specially. When retesting, we no longer saw the icon overlapping text, but the scrollbar could overlay the icon. We decided that was not something we wanted to spend more time on to address.